akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -397,6 +397,9 @@ private[akka] class ShardRegion(
 
   import context.dispatcher
   val retryTask = context.system.scheduler.schedule(retryInterval, retryInterval, self, Retry)
```

Comment:
Since you are now doing the
```
registrationTask = Some(context.system.scheduler.scheduleOnce(nextRegistrationDelay, self, Retry)) 
```

shouldn't this `schedule()` (note that it's not scheduleOnce but periodically) be removed?

Reply:
I don't know. 

https://github.com/akka/akka/blob/28b86379c8889a318378f09d655b5b5714c3dbc7/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala#L572-L582

Some other work is done on `Retry`.

Reply:
Since scheduling is becoming more complicated here now I suggest that we switch to the `Timers` API: https://doc.akka.io/docs/akka/current/actors.html#timers-scheduled-messages

Let's keep the `Retry` as is (but switch to `timers.startPeriodicTimer`).

Add a new `RegisterRetry` for the exponentially increasing registration attempts. `timers.startTimer`
```
case RegisterRetry =>
    if (coordinator.isEmpty) {
        register()
        scheduleNextRegistration()
      }
```

`scheduleNextRegistration` would increase until `> retryInterval` and then the ordinary `Retry` continues